packages/application-shell/src/components/navbar/menu-items.tsx

```diff
@@ -131,42 +122,60 @@ type MenuExpanderProps = {
   isVisible: boolean;
   onClick: MouseEventHandler<HTMLDivElement>;
   isMenuOpen: boolean;
+  isNewNavigationEnabled: boolean;
 };
 const MenuExpander = (props: MenuExpanderProps) => {
   return (
     <li
       key="expander"
-      className={classnames(styles['list-item'], styles.expander, {
+      className={classnames(styles['list-item'], {
         [styles.hidden]: !props.isVisible,
+        [styles['expander-new']]: props.isNewNavigationEnabled,
+        [styles.expander]: !props.isNewNavigationEnabled,
       })}
     >
       <div
         onClick={props.onClick}
-        className={styles['expand-icon']}
+        className={classnames({
+          [styles['expand-icon']]: !props.isNewNavigationEnabled,
+          [styles['expand-icon-new']]: props.isNewNavigationEnabled,
+        })}
         data-testid="menu-expander"
       >
         {/*
           FIXME: define hover effect.
           https://github.com/commercetools/merchant-center-frontend/issues/2216
         */}
-        <ArrowRightIcon color="surface" size="big" />
+        {props.isNewNavigationEnabled ? (
+          props.isMenuOpen ? (
+            <SidebarCollapseIcon color="surface" size="big" />
+          ) : (
+            <SidebarExpandIcon color="surface" size="big" />
+          )
+        ) : (
+          <ArrowRightIcon color="surface" size="big" />
+        )}
```

Comment:
nit: I'm personally fine with that, but I think we should aim to avoid double-nested conditions. Perhaps you could consider using a `do {}` syntax or refactor to use an external function. But it's totally up to you.

Reply:
@kark thanks for pointing that out. I agree it can be written more concisely. 😊 
Let me know if this looks better now with this change - [7198728](https://github.com/commercetools/merchant-center-application-kit/pull/3191/commits/71987284dc9a3172c353905aaab8fd9cda0e28f8)